src/tox/session/env_select.py

```diff
@@ -413,6 +413,9 @@ def iter(  # noqa: A003
             yield name
 
     def ensure_only_run_env_is_active(self) -> None:
+        if not self._state.conf.core["work_dir"].exists():
+            self._state.conf.core["work_dir"].mkdir()
+            (self._state.conf.core["work_dir"] / ".gitignore").write_text("*", encoding="utf-8")
```

Comment:
This doesn't seem right.

Reply:
Ah, I see.

I put the code here because the very next line contains a call to `self._env_name_to_active()`, which ends up creating the `work_dir` if it doesn't already exist. I considered putting the code in `execute()`, but that function wasn't consistently called in the tox subcommands I tested. `ensure_only_run_env_is_active()` was called consistently for the tox subcommands I tested.

The code is duplicated because `info.py` creates a directory and a file that isn't necessarily in the `work_dir` (see the `.tox/` vs `venv/` files listed in the text I posted above). Duplicating the code ensured that a `.gitignore` file existed when the info file wasn't put in a subdirectory of `work_dir`.